lab-doug/route/matchscore-route.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const response = require('../lib/response');
+const Storage = require('../lib/storage');
+const uuid = require('node-uuid');
+var matchScoreStorage = new Storage(__dirname + '/../data');
+
+module.exports = function(router){
+  router.post('/api/matchscore', function(req, res){
+    if(req.body){
+      var postUuid = uuid.v4();
+      req.body.uuid = postUuid;
+      matchScoreStorage.setItem('matchscore', req.body)
+      .then(function(item){
+        response(200, item)(res);
+      }).catch(function(){
+        response(400, 'bad request')(res);
+      });
+    } else {
+      response(404, 'not found')(res);
+    }
+  }).put('/api/matchscore', function(req, res){
+    if(req.body){
+      var postUuid = req.body.uuid;
+      matchScoreStorage.deleteItem('matchscore', postUuid)
+      .then ((uuid, req) => {
+        matchScoreStorage.setItem('matchscore', req.body)
+        .then(function(item) {
+          response(200, item.body)(res);
+        }).catch(function(){
+          response(400, 'bad request')(res);
+        });
+      });
+      response(404, 'not found')(res);
+    }
+  }).get('/api/matchscore', function(req, res){
+    if(!req.url.query.id) return response(400, 'bad request')(res);
+    matchScoreStorage.fetchItem('matchscore', req.url.query.id)
+    .then(function(item){
+      return response(200, item)(res);
+    }).catch(function(){
+      return response(404, 'not found')(res);
+    });
+  })
+  .delete('/api/matchscore', function(req, res){
```

Comment:
I'd be careful to be consistent with your formatting--If you enter down on this line, you should do it for the .get above as well.